### ISSUE TITLE
Convert lng/lat to tile coordinates

### ISSUE THREAD

**Issue by mapsam** [mapbox/vtquery#17]
We are converting tile coordinates to lng/lat, but can we round trip back to tile coordinates? I imagine this could be a reverse engineer of https://github.com/mapbox/vtquery/blob/master/vtquery/reproject.hpp

cc @springmeyer @GretaCB

**Comment by springmeyer** [comment]
I ported `reproject.hpp` from https://github.com/mapbox/vector-tile-js/blob/master/lib/vectortilefeature.js#L137-L145. I'm not aware of how to write the inverse direction. Tagging in @flippmoke for help.

**Comment by flippmoke** [comment]
An example in python - note that this gets you to the zoom level coordinates, to get the tile coordinates offset by `tile_x * extent` and `tile_y * extent`:

```python
def convert_ll_to_vt(lat, lon, tile_z, extent):
    z2 = 1 << tile_z

    # Zoom level coordinates
    lon = ((lon + 180.0) % 360.0)
    if lat > 89.9:
        lat = 89.9
    elif lat < -89.9:
        lat = -89.9
    lat_rad = math.radians(lat)
    zl_x = lon / (360.0 / (extent * z2))
    zl_y = ((extent * z2) / 2) * (1.0 - (math.log(math.tan(lat_rad) + 1.0 / math.cos(lat_rad)) / math.pi))
    zl_x = int(round(zl_x))
    zl_y = int(round(zl_y))
    return [zl_x, zl_y]
```

**Comment by mapsam** [comment]
Resolved in #19
